docs/cip/README.md

```diff
@@ -4,704 +4,2800 @@ Title: Ouroboros Leios - Greater transaction throughput
+
+Additionally, all RBs must follow these content constraints:
+
+1. RBs contain either an EB certificate **or** transactions directly, but
+   **not** both.
+2. All transactions must be valid against the complete ledger state (including
+   certified EBs).
+3. If a node lacks some EB data certified by the chain it is extending, it
+   should issue an RB with no transactions so that the honest chain still
+   definitely gains length.
+
+#### Timing Constraints
+
+The timing constraints work together to maintain Praos' security assumptions
+while enabling higher throughput. These constraints prevent scenarios where
+honest nodes would be forced to delay chain adoption due to missing data. The
+certificate inclusion process (Steps 3-5) involves three sequential timing
+phases:
+
+<a id="equivocation-detection"></a>
+
+**Phase 1: Equivocation Detection ($L_\text{equi}$)**  
+This phase occurs immediately when an RB announces an EB. During this period,
+the network detects any attempts by adversaries to create multiple conflicting
+blocks for the same slot. The equivocation detection mechanism ensures that
+honest nodes can reliably identify and reject equivocating behavior before
+participating in voting.
+
+**Equivocation Attack Model**: An adversary controlling a block production slot
+may attempt to create multiple conflicting EBs and distribute different versions
+to different parts of the network. This could potentially split the honest vote,
+preventing certification of any EB, or worse, allow certification of an
+adversarial EB if honest nodes vote on different versions.
+
+**Detection Mechanism**: The protocol defends against equivocation through a
+multi-step detection process that must accommodate the worst-case propagation
+scenario:
+
+1. **$\Delta_\text{hdr}$**: Initial header propagation - the first (honest or
+   adversarial) RB header reaches all honest nodes
+2. **$\Delta_\text{hdr}$**: Conflicting header propagation - any equivocating
+   header from the same slot reaches honest nodes
+3. **$\Delta_\text{hdr}$**: Equivocation evidence propagation - proof of
+   conflicting headers propagates network-wide, allowing all honest nodes to
+   detect the equivocation
+
+Therefore, $L_\text{equi} \geq 3\Delta_\text{hdr}$ to ensure reliable detection
+before voting begins. This constraint is derived from the network model where
+headers must propagate within $\Delta_\text{hdr}$ to maintain Praos security
+assumptions.
+
+**Security Guarantee**: By waiting $L_\text{equi}$ slots before voting begins,
+the protocol ensures that if any equivocation occurred soon enough to matter,
+all honest nodes will have detected it and will refuse to vote for any EB from
+the equivocating issuer. This prevents adversaries from exploiting network
+partitions to gain unfair advantages in the voting process, as honest nodes will
+only vote for EBs where no equivocation was detected during the detection
+period.
```

Comment:
This is fine, but I think it might differ from the original paper, where detection of equivocation just truncates the propagation of additional equivocated blocks.